packages/app/src/app/hooks/useInterval.ts

```diff
@@ -0,0 +1,21 @@
+// Based on https://overreacted.io/making-setinterval-declarative-with-react-hooks/
+import { useEffect, useRef } from 'react';
+
+export const useInterval = (callback = () => {}, delay: number) => {
+  const savedCallback: { current: any } = useRef();
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  useEffect(() => {
+    function tick() {
+      savedCallback.current();
+    }
+    if (delay !== null) {
+      const id = setInterval(tick, delay);
+      return () => clearInterval(id);
+    }
+    return undefined;
```

Comment:
No need to return `undefined`, since no return is the same

Reply:
That return is simply there to satisfy a linting rule. We could use an ignore instead, though it wouldn't make a difference in how this hook works.